### DISCUSSION TITLE
Will you add Linux support?

### DISCUSSION THREAD

**Discussion by PaullusCarvalho** [T-Troll/alienfx-tools#421] [Q&A]
Hello, how are you doing? So, I'm planning on leaving W11 for good and changing to Ubuntu definetly, will you add linux support for fan control? If not, do you know any similar program that works just as yours do? I much rather just use yours, cause your work is amazing, but i'm still really noob into IT so i don't know if this program of yours works just fine, on linux or not.

**Comment by T-Troll** [comment]
I use Linux for my main duty, but WSL enough for me (and i keep W10 as a main).

So no, Linux support is not planned. However, for fan control, v2 version utilize WMI calls, it's Windows-specific (but some Linux distro have emulation). V1 is direct ACPI, but it should be tuned for every (near every) model - Dell change conventions and method names quite frequently.

Other alternatives may not support your gear - some AW notebooks only have indirect control, but most of them utilize EC-based direct one. Anyway, check GitHub.

**Reply by kaezrr** [reply]
Hey man, I have been looking to switch to linux for a long time but the game mode on my dell g15 5515 ryzen always holds me back.

I love your software and I am a programmer myself, could you tell me what do I need to do to enable gmode on linux? 

Like any commands/scripts, anything you can point me to. You are my last hope please help me out.

**Reply by T-Troll** [reply]
@kaezrr It's possible, but tricky.
First, WMI is not available, but WMI calls mapped to ACPI BIOS method calls in fact.
So....
1. Make ACPI BIOS dump
2. Use my tool [rwdec](https://github.com/T-Troll/rwdec) to decode it and configure out what ACPI method used for AWCC functions
3. Make a script/tool to call it with right parameters (i can help you to configure out which one) - use mine V1 Fan SDK code for reference

BTW, all power/fan controls will be done using this method. But the issue is it's model-depended (that's why i switch to WMI which is not).
